### ISSUE TITLE
[Bug]: Querycoord panic with error `runtime error: index out of range [0] with length 0` after all chaos tests

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#23313]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.0-20230409-7e0f3e92
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):both    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
rootcoord-dml_13_440685973135627463v1] [From=-1] [To=1]
[2023/04/09 23:53:54.188 +00:00] [INFO] [balance/utils.go:54] ["Create Segment task"] [collection=440685973135627479] [replica=440686067459031043] [channel=by-dev-rootcoord-dml_15_440685973135627479v1] [From=-1] [To=11]
[2023/04/09 23:53:54.188 +00:00] [INFO] [balance/utils.go:54] ["Create Segment task"] [collection=440685973135627479] [replica=440686067459031043] [channel=by-dev-rootcoord-dml_14_440685973135627479v0] [From=-1] [To=9]
panic: runtime error: index out of range [0] with length 0

goroutine 91 [running]:
github.com/milvus-io/milvus/internal/querycoordv2/balance.heapQueue.Swap(...)
	/go/src/github.com/milvus-io/milvus/internal/querycoordv2/balance/priority_queue.go:51
container/heap.Pop({0x4275680, 0xc001f51260})
	/usr/local/go/src/container/heap/heap.go:62 +0x45
github.com/milvus-io/milvus/internal/querycoordv2/balance.(*priorityQueue).pop(...)
	/go/src/github.com/milvus-io/milvus/internal/querycoordv2/balance/priority_queue.go:84
github.com/milvus-io/milvus/internal/querycoordv2/balance.(*RowCountBasedBalancer).GenSegmentPlan(0xc0024d9930?, 0xc0001fc900, {{0xc0022857f0, 0x1, 0x1}}, {0xc002285830?, 0x2, 0x2}, 0x3e8)
	/go/src/github.com/milvus-io/milvus/internal/querycoordv2/balance/rowcount_based_balancer.go:197 +0x1c5
github.com/milvus-io/milvus/internal/querycoordv2/balance.(*RowCountBasedBalancer).balanceReplica(0xc001f243c0, 0xc0001fc900)
	/go/src/github.com/milvus-io/milvus/internal/querycoordv2/balance/rowcount_based_balancer.go:186 +0xa50
github.com/milvus-io/milvus/internal/querycoordv2/balance.(*RowCountBasedBalancer).Balance(0xc001f243c0)
	/go/src/github.com/milvus-io/milvus/internal/querycoordv2/balance/rowcount_based_balancer.go:98 +0x27d
github.com/milvus-io/milvus/internal/querycoordv2/checkers.(*BalanceChecker).Check(0xc0006387f8, {0x4271d30, 0xc000890880})
	/go/src/github.com/milvus-io/milvus/internal/querycoordv2/checkers/balance_checker.go:48 +0x76
github.com/milvus-io/milvus/internal/querycoordv2/checkers.(*CheckerController).check(0xc001f28500, {0x4271d30, 0xc000890880})
	/go/src/github.com/milvus-io/milvus/internal/querycoordv2/checkers/controller.go:122 +0x162
github.com/milvus-io/milvus/internal/querycoordv2/checkers.(*CheckerController).Start.func1()
	/go/src/github.com/milvus-io/milvus/internal/querycoordv2/checkers/controller.go:97 +0x132
created by github.com/milvus-io/milvus/internal/querycoordv2/checkers.(*CheckerController).Start
	/go/src/github.com/milvus-io/milvus/internal/querycoordv2/checkers/controller.go:83 +0x8a
```

### Milvus Log

chaos type: pod-failure image tag: 2.2.0-20230409-7e0f3e92 target pod: etcd
failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-kafka-for-release-cron/detail/chaos-test-kafka-for-release-cron/3360/pipeline
log:
[artifacts-etcd-pod-failure-3360-server-logs (1).tar.gz](https://github.com/milvus-io/milvus/files/11187053/artifacts-etcd-pod-failure-3360-server-logs.1.tar.gz)

[artifacts-etcd-pod-failure-3360-pytest-logs.tar.gz](https://github.com/milvus-io/milvus/files/11187054/artifacts-etcd-pod-failure-3360-pytest-logs.tar.gz)

**Comment by zhuwenxing** [comment]
/assign @jiaoew1991 
PTAL

**Comment by yanliang567** [comment]
@aoiasd any updates?

**Comment by aoiasd** [comment]
> @aoiasd any updates?

Panic because pop empty balance queue, seems fixed in #23334 by @weiliu1031
